pkg/prometheus/promcfg_test.go

```diff
@@ -4817,6 +4817,77 @@ remote_write:
   authorization:
     type: Bearer
     credentials: secret
+`,
+		},
+		{
```

Comment:
Can you add a test case with an empty Sigv4 field (e.g. `Sigv4: {}`)?

Reply:
Do you mean that no sigv4 configuration specified by the user will always result in an empty `sigv4: {}` in the Prometheus config file?
If yes, such a test would fail - should it pass?
Does Prometheus requires that section in the configuration file schema?
I see [here](https://github.com/prometheus/prometheus/blob/70f597b0334d48572c4b0c35940c975aef3031c7/docs/configuration/configuration.md#remote_write) that it is mutually exclusive with other authorizations methods.

Reply:
The Prometheus docs say: `To use the default credentials from the AWS SDK, use sigv4: {}` and I suspect that most people would configure it like that. Hence it would be good to ensure that `spec: { remoteWrite: { sigv4: {}, ... } }` translates properly into the Prometheus config file.
If the sigv4 field is `nil` in the Prometheus resource then the the config file shouldn't contain the `sigv4` fiel.

>  I see here that it is mutually exclusive with other authorizations methods.

Correct. From what I can tell, we don't verify either that `Authorization` and `OAuth2` (for example) aren't specified at the same time but it would be good if you add a check for sigv4 at least.

Reply:
Rebased, added the sigv4 tests, and a small validation of the `RemoteWriteSpec` object(s) from the CRD.
I don't expect the final behavior to differ though? We did not change the way we check the configuration is in sync, and that seemed the problem causing the restart...

Reply:
Tested, again now: same problem as before - this seems to be the log from the offending piece of code that triggers the Pod restart
```
level=debug ts=2021-09-07T17:49:29.917839652Z caller=operator.go:1060 component=prometheusoperator msg="StatefulSet key did not match a Prometheus key format" key=default/alertmanager-prom-sigv4-kube-prometheus-alertmanager
```

Reply:
I'll try to rebase and see if it helps